### PR TITLE
Feature/kafkarecordbatch

### PR DESCRIPTION
**Pull request by mehab**
accepting multiple records in snyk analyzer in batch and processing them with multithreading

### CODE REVIEW

File: src/main/java/org/acme/tasks/scanners/SnykAnalysisTask.java [hunk 1]
```diff
@@ -136,49 +136,33 @@ public void analyze(final List<Component> components) {
...
                     }
-
-                    //cache logic end------------------------------------------
-                } catch (UnirestException e) {
-                    handleRequestException(LOGGER, e);
                 }
+                paginatedComponents.nextPage();
+            } catch (InterruptedException ex) {
+                LOGGER.info("Exception occured: " + ex.getMessage());
+                Thread.currentThread().interrupt();
```
Comment (mehab) [inline src/main/java/org/acme/tasks/scanners/SnykAnalysisTask.java:164]: Not very confident on this approach. Followed suggestion from sonar for fix

Comment (nscuro) [reply src/main/java/org/acme/tasks/scanners/SnykAnalysisTask.java:164]: I ran into this a while ago, too. It looks weird, but apparently it's the correct way to do it. This StackOverflow answer explains it very well: https://stackoverflow.com/a/3976377

File: src/main/java/org/acme/tasks/scanners/SnykAnalysisTask.java [hunk 2]
```diff
@@ -136,49 +136,33 @@ public void analyze(final List<Component> components) {
...
+                            temp.add(paginatedList.get(trackComponent));
+                            k += 1;
+                            trackComponent += 1;
+                        }
+                        for (Component component : temp) {
+                            LOGGER.info(component.getName());
                         }
+                        Thread analysisUtil = new Thread(new SnykAnalysisTaskUtil(temp, apiToken));
+                        analysisUtil.start();
+                        analysisUtil.join();
```
Comment (nscuro) [inline src/main/java/org/acme/tasks/scanners/SnykAnalysisTask.java:158]: What is the purpose of launching another thread and then immediately waiting (blocking) for it to complete?

Maybe an `ExecutorService` with [fixed-size thread pool](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Executors.html#newFixedThreadPool(int)) may be a better option?

Comment (mehab) [reply src/main/java/org/acme/tasks/scanners/SnykAnalysisTask.java:155]: Closing this pull request as the changes have already gone in with pull request 39. Will address the review comments from niklas in a different branch regarding the interrupException

File: src/main/java/org/acme/tasks/scanners/SnykAnalysisTask.java (hunk 1 above)
Comment (mehab) [reply src/main/java/org/acme/tasks/scanners/SnykAnalysisTask.java:161]: Closing this pull request as the changes have already gone in with pull request 39. Will address the review comments from niklas in a different branch regarding the interrupException

**Comment by mehab** [comment]
Closing this pull request as the changes have already gone in with pull request 39. Will address the review comments from niklas in a different branch regarding the interrupException

**Pull request closed without merging**